src/java/com/google/cloud/bigquery/dwhassessment/hooks/logger/EventLogger.java

```diff
@@ -0,0 +1,241 @@
+    try {
+      // ScheduledThreadPoolExecutor uses an unbounded queue which cannot be replaced with a
+      // bounded queue.
+      // Therefore, checking queue capacity manually here.
+      if (logWriter.getQueue().size() < queueCapacity) {
+        logWriter.execute(() -> writeEventWithRetries(event));
+      } else {
+        LOG.warn(
+            "Writer queue full ignoring event {} for query {}", hookType, event.get("QueryId"));
+      }
+    } catch (RejectedExecutionException e) {
+      LOG.warn("Writer queue full ignoring event {} for query {}", hookType, event.get("QueryId"));
+    }
+  }
+
+  private DatePartitionedLogger createLogger(String baseDir, HiveConf conf, Clock clock) {
+    if (baseDir == null) {
+      return null;
+    }
+
+    try {
+      return new DatePartitionedLogger(new Path(baseDir), conf, QUERY_EVENT_SCHEMA, clock);
+    } catch (IOException e) {
+      LOG.error("Unable to initialize logger, logging disabled.", e);
+    }
+
+    return null;
+  }
+
+  private String constructFileName() {
+    return "dwhassessment_" + loggerId + "-" + logFileCount + ".avro";
+  }
+
+  private void handleTick() {
+    try {
+      maybeRolloverWriterForDay();
+    } catch (IOException e) {
+      LOG.error("Got IOException while trying to rollover: ", e);
+    }
+  }
+
+  private void maybeRolloverWriterForDay() throws IOException {
+    if (writer == null || !logger.getNow().equals(writerDate)) {
+      if (writer != null) {
+        // Day changes over case, reset the logFileCount.
+        logFileCount = 0;
+        IOUtils.closeQuietly(writer);
+        writer = null;
+      }
+      // increment log file count, if creating a new writer.
+      ++logFileCount;
+      writer = logger.getWriter(constructFileName());
+      writerDate = DatePartitionedLogger.getDateFromDir(writer.getPath().getParent().getName());
+    }
+  }
+
+  private void writeEventWithRetries(GenericRecord event) {
+    for (int retryCount = 0; retryCount <= MAX_RETRIES; ++retryCount) {
+      try {
+        maybeRolloverWriterForDay();
+        writer.writeMessage(event);
+        writer.hflush();
+        return;
+      } catch (IOException e) {
+        // Something wrong with writer – close and reopen.
+        IOUtils.closeQuietly(writer);
+        writer = null;
+        reportRetryState(event, retryCount, e);
+        waitForRetry(retryCount);
+      }
+    }
+  }
+
+  private void reportRetryState(GenericRecord event, int retryCount, IOException writeException) {
+    if (retryCount < MAX_RETRIES) {
+      LOG.warn(
+          "Error writing proto message for query {}, eventType: {}, retryCount: {},"
+              + " error: {} ",
+          event.get("QueryId"),
+          event.get("EventType"),
+          retryCount,
+          writeException.getMessage(),
+          writeException);
+      LOG.trace("Exception", writeException);
+    } else {
+      LOG.error(
+          "Error writing proto message for query {}, eventType: {}: ",
```

Comment:
I would suggest to remove the colon `:` at the end of the error message.

Reply:
Done